sources/controllers/FriendController.ts

```diff
@@ -1,33 +1,178 @@
+        const requestingUser  = getRequestingUser(req);
+
+        const targetUserId = parseInt(req.params.userId, 10);
+        if (isNaN(targetUserId)) {
+            throw Boom.badRequest(`Invalid User id "${req.params.userId}"`);
+        }
+
+        // Check if user exists
+        const targetUser = await this.userRepository.getUserById(targetUserId);
+        if (!targetUser) {
+            throw Boom.notFound(`User id "${req.params.userId}" does not exist`);
+        }
+
+        if (targetUserId === requestingUser.id) {
+            throw Boom.forbidden(`You cannot accept invite from yourself`);
+        }
+
+        // Check existing friendship
+        const existingFriendship = await this.userRepository.getUserFriendship(targetUser.id, requestingUser.id);
+        if (!existingFriendship) {
+            throw Boom.forbidden("Friendship already exist");
+        }
+
+        if (existingFriendship.confirmed) {
+            throw Boom.forbidden("This user is already in your friend list");
+        }
+
+
+        await this.userRepository.updateUserFriendship(requestingUser.id, targetUser.id, true);
         res.status(200).json({
-            message: "Invitation accepted",
+            message: "Invitation Accepted",
         });
     }
 
-    public readonly inviteDecline = async (_req: Request, res: Response) => {
+    public readonly inviteDecline = async (req: Request, res: Response) => {
+        const requestingUser  = getRequestingUser(req);
+
+        const targetUserId = parseInt(req.params.userId, 10);
+        if (isNaN(targetUserId)) {
+            throw Boom.badRequest(`Invalid User id "${req.params.userId}"`);
+        }
+
+        // Check if user exists
+        const targetUser = await this.userRepository.getUserById(targetUserId);
+        if (!targetUser) {
+            throw Boom.notFound(`User id "${req.params.userId}" does not exist`);
+        }
+
+        if (targetUserId === requestingUser.id) {
+            throw Boom.forbidden(`You cannot decline invite from yourself`);
+        }
+
+        // Check existing friendship
+        const existingFriendship = await this.userRepository.getUserFriendship(targetUser.id, requestingUser.id);
+        if (!existingFriendship) {
+            throw Boom.forbidden("Friendship already exist");
+        }
+
+        if (existingFriendship.confirmed) {
+            throw Boom.forbidden("Can't decline an already existing friendship");
+        }
+
+        await this.userRepository.deleteUserFriendship(targetUser.id, requestingUser.id);
         res.status(200).json({
             message: "Invitation declined",
         });
     }
 
-    public readonly getAllFriends = async (_req: Request, res: Response) => {
+    public readonly getAllFriends = async (req: Request, res: Response) => {
+        const requestingUser = getRequestingUser(req);
+
+        const status: true | false | undefined = req.query.params;
```

Comment:
On veut que le parametre se nomme `confirmed`
ex:
- GET /friends # Pour tout (amis et pending)
- GET /friends?confirmed=true # Pour les amis
- GET /friends?confirmed=false # Pour les pending
```suggestion
        const confirmed: true | false | undefined = req.query.confirmed;
```

Aussi, je m'apprete a faire une fonction `parseBool`, tu en auras besoin ici (ça prend un string et renvoit `boolean | undefined`)